raft-tech/TANF-app
# Update API Viewsets

## Summary of Changes
- Updated backend API to support listing users and retrieving individual users
- Added missing features to: user view, stts view, and datafile view
Pull request closes #2466 

## How to Test
```
cd tdrs-frontend && docker-compose up
cd tdrs-backend && docker-compose up
```
- Test User viewset
  - Log in as an OFA System Admin 
    - Go to: `http://localhost:3000/v1/users/` and verify all users are listed
    - Go to: `http://localhost:3000/v1/users/{user_id}/` and verify the correct user is listed
  - Log in as an general user (Data Analyst) 
    - Go to: `http://localhost:3000/v1/users/` and verify only your user is listed
    - Go to: `http://localhost:3000/v1/users/{user_id}/` and verify you can only access your user by ID
- Test STT viewset
  - Go to: `http://localhost:3000/v1/stts/` and verify all stts are displayed with the requested fields
- Test Datafile viewset
  - Go to: `http://localhost:3000/v1/data_files/?stt={stt_code}` and verify all datafiles with that code are listed
  - Note, the exact requested functionality was not implemented here due to a security issue. There will be a follow on bug/ticket to address the issue in a better way.

## Deliverables
_More details on how deliverables herein are assessed included [here](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x] response to user model queries include all relevant fields
+ [x]  response to data_files model queries include all relevant fields
+ [x]  response to stt model queries include all relevant fields

### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
